### PR TITLE
add substring editorial

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

File: solutions/cf-919D.mdx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+while also incrementing $\texttt{dp}[i][l]$ for the current node $i$ where $l$
+represents $s_i$, since we can pick up one more of this letter on our path.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+```cpp
+CodeSnip{Benq Template}
```
Comment (HarryWangATX) [inline solutions/cf-919D.mdx:28]: oops this is still gold 👀

File: solutions/cf-919D.mdx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		F0R(i, N) {
+			rep(26) {
+				ckmax(ret, dp[i][_]);
+			}
+		}
+
+		ps(ret);
+
+	}
+}
```
Comment (HarryWangATX) [inline solutions/cf-919D.mdx:96]: detemplate :(((

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/cf-919D.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+[Official Editorial](https://codeforces.com/blog/entry/57462)
+
+## Explanation
+
+Define $\texttt{dp}[i][j]$ as the maximum number of character $j$ for letter
+$i$. Then, sort the edges topographically in order to perform dynamic
+programming on the DAG. Our transition is simple:
+
+$$
+\texttt{dp}[i][j] = \max \texttt{dp}[k][j] : k \in adj[i]
```
Comment (dongliu0426) [inline solutions/cf-919D.mdx:17]: textttify `adj`

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/cf-919D.mdx (hunk 3 above)
Comment (nwatx) [inline solutions/cf-919D.mdx:17]:
```suggestion
\texttt{dp}[i][j] = \max \texttt{dp}[k][j] : k \in \texttt{adj}[i]
```

File: solutions/cf-919D.mdx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+author: Neo Wang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/57462)
+
+## Explanation
+
+Define $\texttt{dp}[i][j]$ as the maximum number of character $j$ for letter
+$i$. Then, sort the edges topographically in order to perform dynamic
+programming on the DAG. Our transition is simple:
```
Comment (dongliu0426) [inline solutions/cf-919D.mdx:14]: > sort the edges

?

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/cf-919D.mdx (hunk 4 above)
Comment (nwatx) [reply solutions/cf-919D.mdx:14]: > sort the edges topographically

doesn't that mean toposort

Comment (dongliu0426) [reply solutions/cf-919D.mdx:14]: i think nodes?

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**